extensions/amp-selector/0.1/amp-selector.js

```diff
@@ -136,14 +185,30 @@ export class AmpSelector extends AMP.BaseElement {
     if (!el || el.hasAttribute('disabled')) {
       return;
     }
+    /** @type {?Array<string>} */
+    let selectedValues;
     if (el.hasAttribute('selected')) {
       if (this.isMultiple_) {
         this.clearSelection_(el);
-        this.setInputs_();
+        selectedValues = this.setInputs_();
       }
     } else {
       this.setSelection_(el);
-      this.setInputs_();
+      selectedValues = this.setInputs_();
+    }
+
+    // Don't trigger action if selected values haven't changed.
+    if (selectedValues) {
+      // Trigger 'select' event with two data params:
+      // 'targetOption' - option value of the selected or deselected element.
+      // 'selectedOptions' - comma-delimited option values of selected elements.
+      const name = 'select';
+      const detail = {
+        targetOption: el.getAttribute('option'),
+        selectedOptions: selectedValues.join(','),
```

Comment:
Good point, though any separator we'd pick would pose a similar problem. Changed `mutatedAttributesCallback` to allow Array values to avoid string conversion entirely.